src/vs/workbench/contrib/notebook/browser/contrib/statusBar/contributedStatusBarItemController.ts

```diff
@@ -35,9 +35,14 @@ export class ContributedStatusBarItemController extends Disposable implements IN
 	}
 
 	private _updateEverything(): void {
-		this._visibleCells.forEach(cell => cell.dispose());
-		this._visibleCells.clear();
-		this._updateVisibleCells({ added: this._observer.visibleCells, removed: [] });
+		const newCells = this._observer.visibleCells.filter(cell => !this._visibleCells.has(cell.handle));
+		this._visibleCells.forEach(helper => helper.update());
+		this._updateVisibleCells({ added: newCells, removed: [] });
+		const visibleCellHandles = new Set(this._observer.visibleCells.map(item => item.handle));
+		Array.from(this._visibleCells.keys()).filter(handle => !visibleCellHandles.has(handle)).forEach(handle => {
+			this._visibleCells.get(handle)?.dispose();
```

Comment:
Can you have the `_updateVisibleCells` call do this part?

Reply:
Done.